pkg/volume/git_repo/git_repo_test.go

```diff
@@ -311,73 +278,36 @@ func doTestPlugin(scenario struct {
 	return allErrs
 }
 
-func doTestSetUp(scenario struct {
-	name              string
-	vol               *v1.Volume
-	expecteds         []expectedCommand
-	isExpectedFailure bool
-}, mounter volume.Mounter) []error {
+func doTestSetUp(scenario testScenario, mounter volume.Mounter) []error {
 	expecteds := scenario.expecteds
 	allErrs := []error{}
 
-	// Construct combined outputs from expected commands
-	var fakeOutputs []fakeexec.FakeCombinedOutputAction
-	var fcmd fakeexec.FakeCmd
-	for _, expected := range expecteds {
-		if expected.cmd[1] == "clone" {
-			fakeOutputs = append(fakeOutputs, func() ([]byte, error) {
-				// git clone, it creates new dir/files
-				os.MkdirAll(path.Join(fcmd.Dirs[0], expected.dir), 0750)
-				return []byte{}, nil
-			})
-		} else {
-			// git checkout || git reset, they create nothing
-			fakeOutputs = append(fakeOutputs, func() ([]byte, error) {
-				return []byte{}, nil
-			})
+	var commandLog []expectedCommand
+	execCallback := func(cmd string, args ...string) ([]byte, error) {
+		// command is 'git -C <dir> <command> <args>
+		gitCommand := args[2]
+		gitDir := args[1]
+		if gitCommand == "clone" {
+			// Clone creates a directory
+			if scenario.repositoryDir != "" {
+				os.MkdirAll(path.Join(gitDir, scenario.repositoryDir), 0750)
+			}
 		}
+		// add the command to log with de-randomized gitDir
+		args[1] = strings.Replace(gitDir, mounter.GetPath(), "volume-dir", 1)
```

Comment:
Yes, it *should* always return "volume-dir". However, if there is bug in the volume plugin it may be called with a different directory and the test should catch it.